geoh5py/objects/points.py

```diff
@@ -110,53 +102,64 @@ def remove_vertices(
         :param indices: Indices of vertices to remove.
         :param clear_cache: Clear cached data and attributes.
         """
-
-        if self.vertices is None:
-            warnings.warn("No vertices to be removed.", UserWarning)
-            return
-
         if isinstance(indices, list):
             indices = np.array(indices)
 
         if not isinstance(indices, np.ndarray):
             raise TypeError("Indices must be a list or numpy array.")
 
-        if (
-            isinstance(self.vertices, np.ndarray)
-            and np.max(indices) > self.vertices.shape[0] - 1
-        ):
+        if np.max(indices) > self.vertices.shape[0] - 1:
             raise ValueError("Found indices larger than the number of vertices.")
 
         vertices = np.delete(self.vertices, indices, axis=0)
-        self._vertices = None
-        self.vertices = vertices
+        self._vertices = self.validate_vertices(vertices)
         self.remove_children_values(indices, "VERTEX", clear_cache=clear_cache)
+        self.workspace.update_attribute(self, "vertices")
 
-    def copy(
-        self,
-        parent=None,
-        copy_children: bool = True,
-        clear_cache: bool = False,
-        mask: np.ndarray | None = None,
-        **kwargs,
-    ):
+    @property
+    def vertices(self) -> np.ndarray:
```

Comment:
what about a n_vertices property too?
like n_cells in grid objects

Reply:
Yep, already there.